community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/commands/expressions/PointFunction.scala

```diff
@@ -33,14 +33,19 @@ case class PointFunction(data: Expression) extends NullInNullOutExpression(data)
       val map = mapCreator(state.query)
       map.getOrElse("crs", CRS.WGS84.name) match {
         case CRS.Cartesian.name =>
-          val x = map("x").asInstanceOf[Double]
-          val y = map("y").asInstanceOf[Double]
+          if (!map.contains("x") || !map.contains("y")) throw new SyntaxException("A cartesian point must contain 'x' and 'y' coordinates")
```

Comment:
Is this really a `SyntaxException`? It feels to me more like an `InvalidArgumentException`.